### PR TITLE
[BUGFIX release] Upgrade htmlbars to 0.14.14

### PR DESCRIPTION
**Pull request by mmun**
For more information see https://github.com/tildeio/htmlbars/pull/447.

### CODE REVIEW

**Comment by rwjblue** [comment]
I restarted the build after https://github.com/emberjs/ember.js/pull/12879, but it may require a rebase to grab those changes.

**Comment by Dhaulagiri** [comment]
How feasible is it to backport this fix to earlier releases in the 2.x series? (I'm happy to do the actual work)

**Comment by mmun** [comment]
@Dhaulagiri It should be extremely trivial (maybe even just bumping the version in package.json).

I am concerned that you're on an older 2.x release. There should be no reason you are not on the very latest version of Ember. Ember 2.4 being declared an LTS release is even more of a reason to upgrade. Is there a regression that's preventing you from upgrading?

**Comment by rwjblue** [comment]
Agree with @mmun, there should be no blockers from 2.0 through 2.4, if there are we should fix them...

**Comment by Dhaulagiri** [comment]
> It should be extremely trivial (maybe even just bumping the version in package.son)

:+1:

> Is there a regression that's preventing you from upgrading?

There is a longer story here that is probably better off discussed elsewhere (emberconf?). 

tl;dr We have an old karma-based test suite and [non-standard data adapter](https://github.com/heroku/rest-model) that seem to break in some unpredictiable way with every release of Ember.  After prolonged effort we just recently got to 2.0 :tada:.  I was optimistic about quickly getting to 2.4, but our data adapter broke (due to a change to how ember wraps `_super` that I can't find at the moment) and our old test suite won't run at all because [this hack](https://github.com/emberjs/ember.js/issues/10310#issuecomment-95685137) no longer works.  I haven't dug in real hard to fixing either issue due to upgrade fatigue, but I would certainly love to get things sorted out sooner than later.
